fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -2355,6 +2364,13 @@ def cpu_offload(self) -> bool:
         return self.move_params_to_cpu
 
 
+def p_assert(cond: Any, s: Any) -> None:
+    """Used in backward context to make sure error is printed."""
```

Comment:
Doesn't assert already allow you to print an optional message if the condition fails? maybe i am missing something

Reply:
yes, it got swallowed in the backward pass due to the way autograd work.